### DISCUSSION TITLE
Default currency for country does not get set when using IP geolocation

### DISCUSSION THREAD

**Discussion by Ajadaz** [PrestaShop/PrestaShop#35449] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

The shop's currency stays as the default currency when IP geolocation is enabled. It seems that the logic for changing the default currency is only triggered if IP geolocation is not used here https://github.com/PrestaShop/PrestaShop/blob/8.1.4-build/classes/controller/FrontController.php#L327-L360 It is probably possible to extract the currency setting part into a common function or code branch.

### Expected behavior

Having set the default currency in International -> Locations -> Countries, it will be set when using IP geolocation.

### Steps to reproduce

1. Set the shop's default currency in International -> Localization to Euro
2. In International -> Locations -> Countries, set the default currency of United Kingdom to British Pound
3. Go into shop from a UK IP address
4. The default currency will be set to Euro, even though it should be GBP

### PrestaShop version(s) where the bug happened

8.1.4

### PHP version(s) where the bug happened

8.2.7

**Comment by florine2623** [comment]
Hello @Ajadaz ,

I have the same behavior as your.
Although, it is more an Feature request/Improvement that an issue. So I will transfer your issue to the Discussions.

https://github.com/PrestaShop/PrestaShop/assets/16019289/b05b04da-6a91-402c-9d83-70f5697e0228

Thanks!

**Comment by ignas04-2** [comment]
Hi!

I see this more as a bug as well. "Default currency" implies that it should switch according to the country of the shopper, I always wondered why there's no functionality like this and I have to use modules in my shops to achieve this result.

Hopefully it gets changed!
